echopype/echodata/echodata.py

```diff
@@ -133,6 +142,118 @@ def __read_converted(self, converted_raw_path: Optional["PathHint"]):
 
         self.converted_raw_path = converted_raw_path
 
+    def _set_tree(self, tree: DataTree):
+        self._tree = tree
+
+    @classmethod
+    def from_file(cls, converted_raw_path, storage_options=None, open_kwargs={}):
+        echodata = cls(
+            converted_raw_path=converted_raw_path,
+            storage_options=storage_options,
+            open_kwargs=open_kwargs,
+        )
+        echodata._check_path(converted_raw_path)
+        converted_raw_path = echodata._sanitize_path(converted_raw_path)
+        suffix = echodata._check_suffix(converted_raw_path)
+        tree = open_datatree(
+            converted_raw_path,
+            engine=XARRAY_ENGINE_MAP[suffix],
+            **echodata.open_kwargs,
+        )
+        # Change root name to top
+        tree.name = "Top-level"
+
+        echodata._set_tree(tree)
+
+        if isinstance(converted_raw_path, fsspec.FSMap):
+            # Convert fsmap to Path so it can be used
+            # for retrieving the path strings
+            converted_raw_path = Path(converted_raw_path.root)
+        echodata.converted_raw_path = converted_raw_path
+        echodata._load_tree()
+        return echodata
+
+    def _load_tree(self):
+        if self._tree is None:
+            raise ValueError("Datatree not found!")
+
+        for group, value in self.group_map.items():
+            # EK80 data may have a Beam_power group if both complex and power data exist.
+            ds = None
+            try:
+                if value["ep_group"] is None:
+                    node = self._tree
+                else:
+                    node = self._tree[value["ep_group"]]
+
+                ds = self.__get_dataset(node)
+            except ChildResolverError:
+                # Skips group not found errors for EK80 and ADCP
+                ...
+            if group == "top" and hasattr(ds, "keywords"):
+                self.sonar_model = ds.keywords.upper()  # type: ignore
+
+            if isinstance(ds, xr.Dataset):
+                setattr(self, group, ds)
+
+    @property
+    def tree(self):
+        return self._tree
+
+    @property
+    def version_info(self) -> Tuple[int]:
+        if self["Provenance"].attrs.get("conversion_software_name", None) == "echopype":
+            version_str = self["Provenance"].attrs.get("conversion_software_version", None)
+            if version_str is not None:
+                version_num = version_str.split(".")[:3]
+                return tuple([int(i) for i in version_num])
+        return None
+
+    @property
+    def group_paths(self) -> Set[str]:
+        paths = set()
+        root_path = self._tree.pathstr
+        paths.add(root_path)
+        for desc in self._tree.descendants:
+            child_path = desc.pathstr.replace(root_path + "/", "")
+            paths.add(child_path)
+        return paths
```

Comment:
I think you can use Datatree's built-in group function for this:
```
def group_paths(self) -> Set[str]:
    root_path = self._tree.pathstr
    return set([i.replace(root_path + "/", "") for i in self._tree.groups])
```

Reply:
That does seem to work just as well, and it's more compact and easier to understand.

Reply:
Thanks @b-reyes for this suggestion :) Have you tried the direct github code suggestion feature? You should try it so your suggestion can be credited to you! 😄